### PR TITLE
Add support of Auth V2 validator

### PR DESCRIPTION
**Pull request by Kolezhniuk**

### CODE REVIEW

**Review by Kolezhniuk — COMMENTED** [review]
see https://github.com/iden3/contracts/pull/307

File: src/iden3comm/utils/contract-request.utils.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { DID } from '@iden3/js-iden3-core';
+import { IProofService } from '../../proof';
+import { ZeroKnowledgeProofResponse } from '../types';
+
+/**
+ * Retrieves the AuthV2 request ID.
+ *
+ * @returns The AuthV2 request ID.
+ */
+export function getAuthV2RequestId(): number {
```
Comment (vmidyllic) [inline src/iden3comm/utils/contract-request.utils.ts:14]: maybe we can calculate it and put as constant

Comment (vmidyllic) [reply src/iden3comm/utils/contract-request.utils.ts:14]: also we can keep a function and pass circuit id here.

**Review by vmidyllic — DISMISSED** [review]

File: src/iden3comm/utils/contract-request.utils.ts (hunk 1 above)
Comment (Kolezhniuk) [reply src/iden3comm/utils/contract-request.utils.ts:14]: fixed

**Review by vmidyllic — APPROVED** [review]

File: src/proof/proof-service.ts [hunk 2]
```diff
@@ -131,6 +131,15 @@ export interface IProofService {
    */
   generateAuthV2Inputs(hash: Uint8Array, did: DID, circuitId: CircuitId): Promise<Uint8Array>;
 
+  /**
+   * generates auth inputs
+   *
+   * @param {Uint8Array} hash - challenge that will be signed
+   * @param {DID} did - identity that will generate a proof
+   * @returns `Promise<Uint8Array>`
```
Comment (volodymyr-basiuk) [inline src/proof/proof-service.ts:139]: @returns Promise<ZKProof>;

File: src/proof/proof-service.ts [hunk 3]
```diff
@@ -488,6 +497,13 @@ export class ProofService implements IProofService {
     return authInputs.inputsMarshal();
   }
 
+  async generateAuthV2Proof(challenge: Uint8Array, did: DID): Promise<ZKProof> {
```
Comment (volodymyr-basiuk) [inline src/proof/proof-service.ts:500]: /** {@inheritdoc IProofService.generateAuthV2Proof} */

**Review by volodymyr-basiuk — APPROVED** [review]

**Review by volodymyr-basiuk — DISMISSED** [review]

**Review by volodymyr-basiuk — APPROVED** [review]

**Review by daveroga — APPROVED** [review]

**Pull request merged**